pyccel/codegen/printing/ccode.py

```diff
@@ -2494,8 +2527,6 @@ def ccode(expr, filename, assign_to=None, **settings):
         the expression is assigned. Can be a string, ``Symbol``,
         ``MatrixSymbol``, or ``Indexed`` type. This is helpful in case of
         line-wrapping, or for expressions that generate multi-line statements.
-    precision : integer, optional
-        The precision for numbers such as pi [default=15].
     user_functions : dict, optional
         A dictionary where keys are ``FunctionClass`` instances and values are
         their string representations. Alternatively, the dictionary value can
```

Comment:
Not related to this PR, but...

What is the reason for allowing any keyword arguments to this function (which are collected in `**settings`), rather than declaring `(..., user_functions=None, dereference=None)`?

By the way, the signature of the constructor of `CCodePrinter` is
```python
def __init__(self, filename, prefix_module = None):
```
so there are no `user_functions` allowed!

Reply:
i have wondered this for a while myself. There are a few places in the code where this is used. In my opinion it is an anti-pattern. I have been getting rid of them whenever I come across them and am changing a docstring. I'm not even sure that this function is still used

Reply:
Fixed #1791